frontend/lib/src/util/hexToRgba.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Converts a hex color code to an RGBA color.
+ *
+ * @param {string} hex - The hex color code to convert. It can be in the format
+ * of #RGB, #RGBA, #RRGGBB, or #RRGGBBAA.
+ * @returns {[number, number, number, number]} The RGBA representation of the
+ * color. Each value is in the range 0-255.
+ * @throws {Error} If the hex color code is invalid.
+ */
+export function hexToRgba(hex: string): [number, number, number, number] {
```

Comment:
We do have the color2k library already included which [can perform the parsing](https://color2k.com/#parse-to-rgba).

Reply:
Thanks! Lukas also pointed this out, I didn't realize we had this library. Switching over to using that and deleting this!